java/statestore/src/main/java/sleeper/statestore/s3/S3FileReferenceStore.java

```diff
@@ -418,59 +395,21 @@ private String getFilesPath(S3RevisionId revisionId) {
         return stateStorePath + "/files/" + revisionId.getRevision() + "-" + revisionId.getUuid() + "-files.parquet";
     }
 
-    private Record getRecordFromFile(AllReferencesToAFile file) {
-        Record record = new Record();
-        record.put("fileName", file.getFilename());
-        record.put("referencesJson", serDe.collectionToJson(file.getInternalReferences()));
-        record.put("externalReferences", file.getExternalReferenceCount());
-        record.put("lastStateStoreUpdateTime", file.getLastStateStoreUpdateTime().toEpochMilli());
-        return record;
-    }
-
-    private AllReferencesToAFile getFileFromRecord(Record record) {
-        List<FileReference> internalReferences = serDe.listFromJson((String) record.get("referencesJson"));
-        return AllReferencesToAFile.builder()
-                .filename((String) record.get("fileName"))
-                .internalReferences(internalReferences)
-                .totalReferenceCount((int) record.get("externalReferences") + internalReferences.size())
-                .lastStateStoreUpdateTime(Instant.ofEpochMilli((long) record.get("lastStateStoreUpdateTime")))
-                .build();
-    }
-
     private void writeFilesToParquet(List<AllReferencesToAFile> files, String path) throws StateStoreException {
         LOGGER.debug("Writing {} file records to {}", files.size(), path);
-        try (ParquetWriter<Record> recordWriter = ParquetRecordWriterFactory
-                .createParquetRecordWriter(new Path(path), FILE_SCHEMA, conf)) {
-            for (AllReferencesToAFile file : files) {
-                recordWriter.write(getRecordFromFile(file));
-            }
-        } catch (IOException e) {
-            throw new StateStoreException("Failed writing files", e);
-        }
+        stateStoreFileUtils.saveFiles(path, files.stream());
         LOGGER.debug("Wrote {} file records to {}", files.size(), path);
     }
 
     private List<AllReferencesToAFile> readFilesFromParquet(String path) throws StateStoreException {
         LOGGER.debug("Loading file records from {}", path);
-        List<AllReferencesToAFile> files = new ArrayList<>();
-        try (ParquetReader<Record> reader = fileReader(path)) {
-            ParquetReaderIterator recordReader = new ParquetReaderIterator(reader);
-            while (recordReader.hasNext()) {
-                files.add(getFileFromRecord(recordReader.next()));
-            }
-        } catch (IOException e) {
-            throw new StateStoreException("Failed loading files", e);
-        }
+        List<AllReferencesToAFile> files = stateStoreFileUtils.loadFiles(path)
+                .referencedAndUnreferenced()
+                .collect(Collectors.toList());
```

Comment:
It looks unnecessary to gather a StateStoreFiles object here and then throw it away. It's also changing the sort order of the files, which is why the test is failing.